src/readbin/compare.ml

```diff
@@ -0,0 +1,119 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+exception No_unstripped_file
+
+let print_source = function
+  | `bw -> print_endline "bw"
+  | _ -> print_endline "others"
+
+let source = [
+  "BW", `bw;
+  "SymTbl", `symtbl;
+  "User", `user;
+  "Ida", `ida;
+]
+
+let tool : _ Term.t =
+  let doc = "The tool that we are going to evaluate. It can be BW or Ida." in
+  Arg.(value & pos 1 (enum source) `bw & info [] ~docv:"tool" ~doc)
+
+let gt : _ Term.t =
+  let doc = "The ground truth source. One can direct to a user file or use the symbol
+  table from unstripped binary (-u required)." in
+  Arg.(value & pos 2 (enum source) `symtbl & info [] ~docv:"gt" ~doc)
+
+let symsfile : string option Term.t =
+  let doc = "The symbol table of binaries. This requires the binaries to use this file as symbols source." in
+  Arg.(value & opt (some non_dir_file) None & info ["syms"; "s"]
+         ~docv:"syms" ~doc)
+
+let unstrip_bin : string option Term.t =
```

Comment:
but the `unstrip_bin` is required only when the ground truth is assigned to `SymTbl`, which is the symbol table. Is there a way to set required with a specific value for `gt`? Currently what I did is to check if `unstrip_bin` is not none when `gt` is `SymTbl.`